### PR TITLE
Makeitpasstest

### PR DESCRIPTION
**Pull request by ilovemane**

### CODE REVIEW

File: R/generateSyntheticCohortSet.R [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#' @param seed random seed
+#' @param cdmVersion version number of the cdm
+#'
+#' @return A cdm reference with the mock tables
+#' @export
+#'
+generateSyntheticCohortSet <- function(cdm,
+                                       tableName = "cohort",
+                                       numberCohorts = 2,
+                                       cohortName = paste0("cohort_", cohortId),
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:36]:
```
cohortName = paste0("cohort_", seq_len(numberCohorts))
```

File: R/generateSyntheticCohortSet.R [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#' @param cohortName name of the cohort within the table
+#' @param recordPerson the expected number of record per person
+#' @param seed random seed
+#' @param cdmVersion version number of the cdm
+#'
+#' @return A cdm reference with the mock tables
+#' @export
+#'
+generateSyntheticCohortSet <- function(cdm,
+                                       tableName = "cohort",
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:34]: name

File: R/generateSyntheticCohortSet.R [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                                       numberCohorts = 2,
+                                       cohortName = paste0("cohort_", cohortId),
+                                       recordPerson = 1,
+                                       seed = NULL,
+                                       cdmVersion = attr(cdm, "cdm_version")) {
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:39]: get from cdm

File: R/generateSyntheticCohortSet.R [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+#'
+#' @return A cdm reference with the mock tables
+#' @export
+#'
+generateSyntheticCohortSet <- function(cdm,
+                                       tableName = "cohort",
+                                       numberCohorts = 2,
+                                       cohortName = paste0("cohort_", cohortId),
+                                       recordPerson = 1,
+                                       seed = NULL,
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:38]: should we have a default seed?

File: R/generateSyntheticCohortSet.R [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+  # initial checks
+  # checkInput(
+  #   cdm = cdm, tableName = tableName, cohortId = cohortId,
+  #   cohortName = cohortName, numberRows = numberRows, seed = seed,
+  #   cdmVersion = cdmVersion,
+  #   .options = list(cdmRequiredTables = c("person", "observation_period"))
+  # )
+
+  if (!is.null(seed)) {
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:49]: remove if

File: R/generateSyntheticCohortSet.R [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    set.seed(seed = seed)
+  }
+
+  #generate synthenic cohort id
+  cohortId = 1:numberCohorts
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:54]: use seq_len(numberCohorts)

File: R/generateSyntheticCohortSet.R [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+  #number of rows per cohort
+  numberRows <- recordPerson*(cdm$person |> nrow()) |> round()
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:57]: use tallly so if it is in a database it also works

File: R/generateSyntheticCohortSet.R [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  cohort <- list()
+  if (length(numberRows) == 1) {
+    numberRows <- rep(numberRows, length(cohortId))
+  }
+  for (i in seq_along(cohortId)) {
+    num <- numberRows[i]
+    cohort[[i]] <- dplyr::tibble(
+      cohort_definition_id = cohortId[i],
+      subject_id = sample(
+        x = cdm$person$person_id, size = num, replace = TRUE
```
Comment (catalamarti) [inline R/generateSyntheticCohortSet.R:69]: can we use pull("person_id")

**Review by catalamarti — APPROVED** [review]

**Comment by ilovemane** [comment]
@catalamarti I have added some tests and this mock can now generate all the standard clinical table, as well as person and observation_period. It would be nice if you can merge this to main, until it gets too big.

**Pull request merged**